playbooks/test.yml

```diff
@@ -0,0 +1,74 @@
+---
+
+- name: automation-broker-apb test playbook
+  hosts: localhost
+  gather_facts: false
+  connection: local
+  tasks:
+    # Test crd
+    - name: Provision test
+      include_role:
+        name: automation-broker-apb
+      vars:
+        apb_action: provision
+        create_broker_namespace: true
+        broker_dao_type: crd
+
+    - name: Wait for clusterservicebroker to become available
+      command: kubectl get clusterservicebrokers automation-broker -o go-template={%raw%}'{{ range .status.conditions }}{{ if eq .type "Ready" }}{{ .status }}{{ end }}{{ end }}'{%endraw%}
+      register: broker_ready
+      retries: 60
+      delay: 10
+      until: broker_ready.stdout == 'True'
+
+    - name: Deprovision test
+      include_role:
+        name: automation-broker-apb
+      vars:
+        apb_action: deprovision
+        create_broker_namespace: true
+        broker_dao_type: crd
+
+    - name: Wait for namespace to be destroyed
+      vars:
+        namespace_lookup: "{{ lookup('k8s', kind='Namespace', resource_name='automation-broker') }}"
+      set_fact:
+        namespace: "{{ namespace_lookup }}"
+      retries: 10
+      delay: 5
+      until:
+        - not namespace_lookup
+
+    # Test etcd
+    - name: Provision test
+      include_role:
+        name: automation-broker-apb
+      vars:
+        apb_action: provision
+        create_broker_namespace: true
+        broker_dao_type: etcd
+
+    - name: Wait for clusterservicebroker to become available
+      command: kubectl get clusterservicebrokers automation-broker -o go-template={%raw%}'{{ range .status.conditions }}{{ if eq .type "Ready" }}{{ .status }}{{ end }}{{ end }}'{%endraw%}
+      register: broker_ready
+      retries: 60
+      delay: 10
+      until: broker_ready.stdout == 'True'
+
+    - name: Deprovision test
+      include_role:
+        name: automation-broker-apb
+      vars:
+        apb_action: deprovision
+        create_broker_namespace: true
+        broker_dao_type: etcd
+
+    - name: Wait for namespace to be destroyed
+      vars:
+        namespace_lookup: "{{ lookup('k8s', kind='Namespace', resource_name='automation-broker') }}"
+      set_fact:
+        namespace: "{{ namespace_lookup }}"
```

Comment:
not sure you want to name this `namespace`, since that's a commonly used APB parameter. Maybe `namespace_status` or something like that?